src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts

```diff
@@ -75,9 +223,92 @@ g.test('t2t_non_compressed_color_formats')
             srcTextureSize: { width: 32, height: 32, depth: 1 },
             dstTextureSize: { width: 63, height: 61, depth: 1 },
           },
+        ])
+      )
+      .combine(poptions('format', kRegularTextureFormats))
+      .combine(
+        poptions('copyBoxOffsets', [
+          // From (0, 0) of src to (0, 0) of dst.
           {
-            srcTextureSize: { width: 32, height: 32, depth: 3 },
-            dstTextureSize: { width: 32, height: 32, depth: 3 },
+            srcOffset: { x: 0, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (0, 0) of src to (1, 0) of dst.
+          {
+            srcOffset: { x: 0, y: 0, z: 0 },
+            dstOffset: { x: 1, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (0, 0) of src to (0, 1) of dst.
+          {
+            srcOffset: { x: 0, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 1, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (1, 0) of src to (0, 0) of dst.
+          {
+            srcOffset: { x: 1, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (0, 1) of src to (0, 0) of dst.
+          {
+            srcOffset: { x: 0, y: 1, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: 0, depth: 0 },
+          },
+          // From (1, 0) of src to (0, 0) of dst, and the copy extent will not cover the last
+          // column of both source and destination texture.
+          {
+            srcOffset: { x: 1, y: 0, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: -1, height: 0, depth: 0 },
+          },
+          // From (0, 1) of src to (0, 0) of dst, and the copy extent will not cover the last
+          // row of both source and destination texture.
+          {
+            srcOffset: { x: 0, y: 1, z: 0 },
+            dstOffset: { x: 0, y: 0, z: 0 },
+            copyExtent: { width: 0, height: -1, depth: 0 },
+          },
+        ])
+      )
+      .combine(poptions('srcCopyLevel', [0, 3]))
+      .combine(poptions('dstCopyLevel', [0, 3]))
+  )
+  .fn(async t => {
+    const { textureSize, format, copyBoxOffsets, srcCopyLevel, dstCopyLevel } = t.params;
+
+    t.DoCopyTextureToTextureTest(
+      textureSize.srcTextureSize,
+      textureSize.dstTextureSize,
+      format,
+      copyBoxOffsets,
+      srcCopyLevel,
+      dstCopyLevel
+    );
+  });
+
+// TODO(jiawei.shao@intel.com): support all WebGPU texture formats
+g.test('t2t_2d_array_non_compressed_color_textures')
```

Comment:
nit: name these two so they are siblings in the tree. e.g.
`color_textures,non_compressed,array`
`color_textures,non_compressed,non_array`
(And can remove "t2t" because it's redundant with the filename.)